### PR TITLE
Add EIP: Oauth2.0 Authentication via Ethereum

### PR DESCRIPTION
**Pull request by xinbenlv**
WIP NOT READY FOR MERGE.

Creating a Oauth2.0 via Ethereum ERC. This is going to take a while and so please wait until a signal before reviewing

### CODE REVIEW

**Comment by xinbenlv** [comment]
WIP

**Comment by xinbenlv** [comment]
WIP

**Comment by xinbenlv** [comment]
WIP

**Comment by xinbenlv** [comment]
WIP

**Comment by Pandapip1** [comment]
@xinbenlv if you're interested, I would be love to work on this with you.

File: EIPS/eip-5735.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## Reference Implementation
+An optional section that contains a reference/example implementation that people can use to assist in understanding or implementing this specification.  If the implementation is too large to reasonably be included inline, then consider adding it as one or more files in `../assets/eip-####/`.
+
+## Security Considerations
+All EIPs must contain a section that discusses the security implications/considerations relevant to the proposed change. Include information that might be important for security discussions, surfaces risks and can be used throughout the life cycle of the proposal. E.g. include security-relevant design decisions, concerns, important discussions, implementation-specific guidance and pitfalls, an outline of threats and risks and how they are being addressed. EIP submissions missing the "Security Considerations" section will be rejected. An EIP cannot proceed to status "Final" without a Security Considerations discussion deemed sufficient by the reviewers.
```
Comment (Pandapip1) [inline EIPS/eip-5735.md:38]:
```suggestion
Needs discussion.
```

File: EIPS/eip-5735.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages.
+
+## Backwards Compatibility
+All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.
+
+## Test Cases
+Test cases for an implementation are mandatory for EIPs that are affecting consensus changes.  If the test suite is too large to reasonably be included inline, then consider adding it as one or more files in `../assets/eip-####/`.
+
+## Reference Implementation
+An optional section that contains a reference/example implementation that people can use to assist in understanding or implementing this specification.  If the implementation is too large to reasonably be included inline, then consider adding it as one or more files in `../assets/eip-####/`.
```
Comment (Pandapip1) [inline EIPS/eip-5735.md:35]:
```suggestion
```

File: EIPS/eip-5735.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+
+
+## Rationale
+The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages.
+
+## Backwards Compatibility
+All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.
```
Comment (Pandapip1) [inline EIPS/eip-5735.md:29]:
```suggestion
No backward compatibility issues found. 
```

File: EIPS/eip-5735.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+## Motivation
+Oauth2.0 is one of the most widely used authentication and authorization schemes. Supporting OAuth2.0 paves way for massive ID onramp adoptions for web2 users.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+
+
+## Rationale
+The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages.
```
Comment (Pandapip1) [inline EIPS/eip-5735.md:26]:
```suggestion
TBD
```

File: EIPS/eip-5735.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
+eip: 5735
+title: Oauth2.0 Authentication via Ethereum
+description: An authentication scheme for Oauth2.0 via Ethereum.
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: <URL>
```
Comment (Pandapip1) [inline EIPS/eip-5735.md:6]: Add discussions-to link.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Comment by Pandapip1** [comment]
I removed the optional sections and added placeholders where needed.

Once the spec and discussion link are filled out, this is ready to be merged.

**Pull request closed without merging**